### PR TITLE
Lh/fix doubleclick

### PR DESCRIPTION
**Pull request by lucyhutcheson**
**Related Issue(s):**

- #43 

**Proposed Changes:**

1.  Add API_MAX_ITEMS
2.  Show results display for mosaic view

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/Element84/filmdrop-ui/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: src/utils/datetime.js [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
-  const fromDate = convertDate(dateTimeRef[0])
-  const toDate = convertDate(dateTimeRef[1])
+// date used for STAC API request for scene view
+export const convertDateForURL = (dateTimeRef) => {
+  const fromDate = convertDateTime(dateTimeRef[0])
+  const toDate = convertDateTime(dateTimeRef[1])
 
-  const datetime = `${fromDate}/${toDate}`
-  return urlEscaped ? encodeURIComponent(datetime) : datetime
+  return encodeURIComponent(`${fromDate}/${toDate}`)
```
Comment (philvarner) [inline src/utils/datetime.js:27]: export const convertDateForURL = (dateTimeRef) => encodeURIComponent(convertDate(dateTimeRef))

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -204,13 +216,29 @@ const Search = () => {
...
         const feature = _searchResults.features[f]
         const featureBounds = setupBounds(feature.bbox)
 
+        // preserve current image if clicking on the same image
+        if (clickBounds.intersects(currentImageClickBoundsRef.current)) {
+          currentImageClickedRef.current = true
+        } else {
+          currentImageClickedRef.current = false
+          currentImageClickBoundsRef.current = clickBounds
+          if (clickedFootprintsHighlightLayer)
```
Comment (philvarner) [inline src/components/Search/Search.js:238]: same w/ {}, there are a few other places with this too.

File: src/components/Search/Search.js [hunk 3]
```diff
@@ -246,18 +274,59 @@ const Search = () => {
...
+      .join('&')
+
+    dispatch(setSearchParameters(searchParamsStr))
+
+    const searchURL = `${process.env.REACT_APP_STAC_API_URL}/search?${searchParamsStr}`
+
+    const response = await fetch(searchURL)
+    if (!response.ok) {
+      const message = `An error has occurred: ${response.status}`
+      throw new Error(message)
```
Comment (philvarner) [inline src/components/Search/Search.js:326]:
```suggestion
      throw new Error(`An error has occurred: ${response.status}`)
```

I usually only put the error message into a variable if I need to do two things with it, like log and use it in the Error msg.

File: src/components/Search/Search.js [hunk 4]
```diff
@@ -362,7 +395,7 @@ const Search = () => {
         const tileBounds = setupBounds(json.bbox)
 
         L.tileLayer(
-          `${tilerURL}/stac/tiles/{z}/{x}/{y}.png?url=${featureURL}${tilerParams}`,
+          `${tilerURL}/stac/tiles/{z}/{x}/{y}.png?url=${featureURL}&${tilerParams}`,
```
Comment (philvarner) [inline src/components/Search/Search.js:398]: 👍🏻

**Review by philvarner — APPROVED** [review]
A few minor issues I'd like to see addressed, but let's ship it

File: src/utils/datetime.js (hunk 1 above)
Comment (lucyhutcheson) [reply src/utils/datetime.js:27]: Completed in next PR

File: src/components/Search/Search.js (hunk 2 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:238]: Completed in next PR

File: src/components/Search/Search.js (hunk 3 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:326]: Completed in next PR

**Pull request merged**